dotnet-statsig/src/Statsig/Lib/ConcurrentHashSet.cs

```diff
@@ -0,0 +1,128 @@
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Statsig.Server.Lib
+{
+    class ConcurrentHashSet<T>: IDisposable
```

Comment:
Is this really necessary over using ConcurrentDictionary? If so, why not just using locks in the `Add`/`Remove`/`Contains` function in `IDList`?

Reply:
The hashset saved over 100mb (out of 800), so i thought it was worth it.
We could have just moved locks to IDList - didn't think of that, since the hashset implementation was available already.  I don't have a strong preference either way.